pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -1826,6 +1826,13 @@
     )
     private boolean exposePreciseBacklogInPrometheus = false;
 
+    @FieldContext(
+            category = CATEGORY_METRICS,
+            doc = "Enable expose the backlog size fro each subscription when generating stats.\n" +
+                    " Locking is used for fetching the status so default to false."
+    )
+    private boolean exposeSubscriptionBacklokSizeInPrometheus = false;
```

Comment:
```suggestion
    private boolean exposeSubscriptionBacklogSizeInPrometheus = false;
```

Reply:
will fix.

Reply:
I think we don't need to introduce a new configuration for this metric? If users enabled the `exposeTopicLevelMetricsInPrometheus`, this metric should be exposed? If it will bring a lot of expenses, we should consider adding a new configuration such as `exposePreciseBacklogInPrometheus`. What do you think @zymap @MarvinCai?

Reply:
I think it is actually quiet expensive, it's using existing [estimateBacklogFromPosition(PositionImpl pos)](https://github.com/apache/pulsar/blob/master/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java#L1120) to get the backlog size, which requires locking the whole MLedger object. And this will happen for every subscription, so periodic metric generation with this subscription backlog size could affect performance, so I think we should add a flag for it and disable by default.